examples/go-client/README.md

```diff
@@ -20,17 +17,56 @@ Auth API clients must perform two-way authentication using x509 certificates:
 2. They must offer their x509 certificate, which has been previously issued
    by the auth sever.
 
-Start up a teleport auth server and then run the following commands to create an `api-user` with a signed certificate. Make sure to run this from the `go-client` directory for proper output.
+There are a few simple ways to generate the certificates needed to authenticate the client, each of which is detailed in this demo.
+
+### Authorization
+
+The Server will authorize requests based on the user associated with the certs used to authenticate the Client. 
+
+Therefore, to use the API client, you need to create a user and any roles it may need for your use case. The client will act as that user and have access to everything defined by the user's role(s).
+
+Note: role based access control is an enterprise feature. All users will be treated 
+as the `admin` role in non-enterprise instances of Teleport. 
+
+### Run the Demo
+
+First, create the `access-admin` user and role using the commands below.
 
 ```bash
-$ tctl create -f api-admin.yaml
-$ mkdir -p certs
-$ tctl auth sign --format=tls --ttl=87600h --user=api-admin --out=certs/api-admin
+$ tctl create -f access-admin.yaml
+$ tctl users add access-admin --roles=access-admin
 ```
 
-### Demo
+Second, Replace the address `proxy.example.com:3025` with the address of your auth server.
+
+Third, choose one of the authentication methods.
+
+1. tsh profile (default):
+
+   ```bash 
+   $ tsh login --user=access-admin
+   ```
+
+2. identity file:
+
+   In `main.go` replace `client.ProfileCreds()` (line 33), with `client.IdentityFileCreds("identity_file_path")`.
```

Comment:
Link to the line in main.go directly? This has the downside that the reference might become invalid if changes are not synced between Go code and the README.
Show a snippet from the example to make it visually helpful.